core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -931,9 +931,8 @@ public void addEnv(String key, String value) {
      */
     @Override
     public void addFileSystemBind(final String hostPath, final String containerPath, final BindMode mode, final SelinuxContext selinuxContext) {
-
         final MountableFile mountableFile = MountableFile.forHostPath(hostPath);
-        binds.add(new Bind(mountableFile.getResolvedPath(), new Volume(containerPath), mode.accessMode, selinuxContext.selContext));
```

Comment:
`resolvedPath` unpacks the resource if it is in a JAR file, we can't replace it with `getFilesystemPath()`

Reply:
Both `getResolvedPath` and `getFilesystemPath` preforms resource unpack (to be honest idk why both).
In my case (#2508) i'm trying to bind any resource from classpath (it can be located either in src/test/resources/example.foo or tescontainers-1.13.0.jar/docker-java.properties eg) with `withClasspathResourceMapping`. Then this method just call `addFileSystemBind` with already resolved resource path ("src/test/resources/example.foo" transforms into "c:/project/src/test/resources/example.foo" and  "tescontainers-1.13.0.jar/docker-java.properties" transforms into "%TEMP%/.testcontainers-tmp-15874678656018686467"): https://github.com/testcontainers/testcontainers-java/blob/master/core/src/main/java/org/testcontainers/containers/GenericContainer.java#L1177
Due javadoc on [`addFileSystemBind`](https://github.com/testcontainers/testcontainers-java/blob/master/core/src/main/java/org/testcontainers/containers/Container.java#L107) and intuition we should deal only with local paths here. So i assume, that here we can convert window path to linux (what MountableFile#resolveFilesystemPath do)

Reply:
I remember touching `resolveFilesystemPath()` when first getting docker-compose support working and it was already a bit crufty back the. Since it internally calls `getResourcePath()`, it will also unpack JARs.

The difference to `getResolvedPath()` is only in the Windows case using `PathUtils.createMinGWPath(result)`.

Whatever we decide, we should probably deprecate one of those methods and make the naming potentially clearer.